Obsidian/Commands/MainCommandModule.cs

```diff
@@ -331,6 +331,62 @@ public Task StopAsync(CommandContext ctx)
         return Task.CompletedTask;
     }
 
+    [Command("ban")]
+    [CommandInfo("Bans a user.", "/ban <user> <duration>")]
+    [RequirePermission(permissions: "obsidian.ban")]
+    public async Task BanAsync(CommandContext ctx, string username, int duration)
+    {
+        var server = (Server)ctx.Server;
+        var player = server.GetPlayer(username);
+        server.Bans.AddBan(player, duration);
+
+        if (ctx.IsPlayer)
+        {
+            await player.KickAsync($"{username} was banned by {ctx.Player.Username}");
+        }
+        else
+        {
+            await player.KickAsync($"{username} was banned by Console");
+        }
+    }
+
```

Comment:
Why is duration mandatory? It should be possible to permanently ban players. Consider making Duration nullable.